### DISCUSSION TITLE
Website URLs no longer allowed to be relative?

### DISCUSSION THREAD

**Discussion by nealpaze** [code4recovery/12-step-meeting-list#1591] [General]
I went in to edit a meeting that is part of a group and suddenly the relative URLs seem to not work now. I made no change to the URL, I was simply removing a contact from a group and it won't let me save it. We have 212 meetings right now and having to keep pasting in a full URL for meetings that already have a relative URL will be very troublesome. And since we have had these URLs in this format for years, I would love for this to revert back.

https://www.nwarkaa.org
![Neal C12-16-2024 at 08 10@2x](https://github.com/user-attachments/assets/bc293ba7-a17a-44d6-9eed-6ecf63687783)
Version 3.16.7

I confirmed that adding a fully qualified URL such as "https://www.nwarkaa.org/farmington/#turning-point" vs. "/farmington/#turning-point" works, but again, this will just make doing updates much more troublesome. Since this has never been an issue before, it would be great if this were reverted.

Thanks for your time and consideration on this.

**Comment by joshreisner** [comment]
hi @nealpaze apologies about the inconvenience, but this field is really supposed to be a fully-qualified URL, so it can be properly shared with other service entities if needed. we always had `https://` placeholders to let people know, but recently changed the field from an `input type="text"` to an `input type="url"` which is why you're seeing the error now

there are two easy ways to update these to be full URLS: 
1. you could export your meetings as a CSV, make the changes there, and reimport
2. you could run this SQL statement. i just tested this locally and it worked for me. you might want to make a sql backup first however just to be safe. also note that some wordpress sites don't use the default `wp_` as the table prefix, so you might need to adjust the table names (in two places)

```sql
update
	wp_postmeta m
set
	m.meta_value = concat('https://www.nwarkaa.org', m.meta_value)
where 
	m.meta_key in ('website', 'website_2') 
	and m.post_id in (select p.id from wp_posts p where p.post_type in ('tsml_meeting', 'tsml_group')) 
	and m.meta_value like '/%'
```

**Comment by nealpaze** [comment]
Thank you @joshreisner. We do share our feed and it didn't occur to me how that would affect outside services. It also didn't occur to me that some groups or meetings might have their own site which is different than our area set up which would also necessitate a fully qualified URL. I appreciate the explanation!

Also, after changing the table names to match our hosting system's schema, that SQL ran like a dream. I really appreciate the help! I am not always the only one who manages meeting changes, so this will make it much easier on everyone involved. Thanks for your service!

**Reply by joshreisner** [reply]
glad it worked 😅
